packages/cli/src/commands/init/init.ts

```diff
@@ -501,25 +476,32 @@ async function createProject(
-  // | Arguments                                                        | Template | React Native |
-  // +==================================================================+==========+==============+
-  // | <None>                                                           | latest   | latest       |
-  // +------------------------------------------------------------------+----------+--------------+
-  // | --version 0.75.0                                                 | 0.75.0   | 0.75.0       |
-  // +------------------------------------------------------------------+----------+--------------+
-  // | --template @react-native-community/template@0.75.1               | 0.75.1   | latest       |
-  // +------------------------------------------------------------------+----------+--------------+
-  // | --template @react-native-community/template@0.75.1 --version 0.75| 0.75.1   | 0.75.x       |
-  // +------------------------------------------------------------------+----------+--------------+
+  // +==================================================================+==========+===================+
+  // | Arguments                                                        | Template |   React Native    |
+  // +==================================================================+==========+===================+
+  // | <None>                                                           | 0.74.x   | 0.74.5 (latest)   |
+  // +------------------------------------------------------------------+----------+-------------------+
+  // | --version next                                                   | 0.75.x   | 0.75.0-rc.1 (next)|
+  // +------------------------------------------------------------------+----------+-------------------+
+  // | --version 0.75.0                                                 | 0.75.x   | 0.75.0            |
+  // +------------------------------------------------------------------+----------+-------------------+
+  // | --template @react-native-community/template@0.75.1               | 0.75.1   | latest            |
+  // +------------------------------------------------------------------+----------+-------------------+
+  // | --template @react-native-community/template@0.75.1 --version 0.75| 0.75.1   | 0.75.x            |
+  // +------------------------------------------------------------------+----------+-------------------+
   //
   // 1. If you specify `--version 0.75.0` and `@react-native-community/template@0.75.0` is *NOT*
   // published, then `init` will exit and suggest explicitly using the `--template` argument.
   //
   // 2. `--template` will always win over `--version` for the template.
   //
   // 3. For version < 0.75, the template ships with react-native.
-  const templateUri = await createTemplateUri(options, version);
+  const v = semver.parse(version);
+  if (v == null) {
+    throw new Error(`--version=${version} isn't valid SEMVER`);
+  }
+  const templateVersion = `${v.major}.${v.minor}`;
```

Comment:
IMHO this can't work as it is.
This will return `0.75` and will fail.

Moreover it's breaking the `createTemplateUri` logic here:

```
  if (options.platformName) {
    logger.debug('User has specified an out-of-tree platform, using it');
    return `${options.platformName}@${version}`;
  }
```

You should pass the version as the user specifies it and not attempt to parse it